pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -384,6 +444,23 @@ func updateVersion(ch chan<- prometheus.Metric) {
 	)
 }
 
+type vmiMetrics struct {
```

Comment:
I think you actually don't need this and all these Desc can be local. Please ignore this if I missed some cases.

Otherwise, it seems good to me. Please just squash commits and ping me :+1: